### PR TITLE
Refactor GAV, GA entities to allow easier manipullation with them

### PR DESCRIPTION
**Pull request by jbartece**

### CODE REVIEW

**Comment by jbartece** [comment]
retest this please

**Comment by jbartece** [comment]
retest this please

File: communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-import java.net.URLConnection;
-import java.util.List;
-
-import javax.xml.bind.JAXBContext;
-import javax.xml.bind.JAXBException;
-import javax.xml.bind.Unmarshaller;
-
 import org.jboss.da.common.util.Configuration;
 import org.jboss.da.common.util.ConfigurationParseException;
+import org.jboss.da.communication.CommunicationException;
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java]: Just changed order.

File: communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
-import org.jboss.da.communication.CommunicationException;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.List;
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java]: Just changed order.

File: communication/src/main/java/org/jboss/da/communication/aprox/api/AproxConnector.java [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-import java.util.List;
 import org.jboss.da.communication.CommunicationException;
-
-import org.jboss.da.communication.aprox.model.GA;
-import org.jboss.da.communication.aprox.model.GAV;
 import org.jboss.da.communication.aprox.model.GAVDependencyTree;
+import org.jboss.da.communication.model.GA;
+import org.jboss.da.communication.model.GAV;
+
+import java.util.List;
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/aprox/api/AproxConnector.java]: Just changed order.

**Comment by jbartece** [comment]
We should define the order of the packages in the imports in IDE

File: reports-backend/src/main/java/org/jboss/da/reports/backend/impl/DependencyTreeGeneratorImpl.java [hunk 4]
```diff
@@ -1,14 +1,14 @@
 package org.jboss.da.reports.backend.impl;
 
+import org.jboss.da.communication.CommunicationException;
```
Comment (janinko) [inline reports-backend/src/main/java/org/jboss/da/reports/backend/impl/DependencyTreeGeneratorImpl.java]: Just changed order.

File: reports-backend/src/main/java/org/jboss/da/reports/backend/impl/VersionFinderImpl.java [hunk 5]
```diff
@@ -1,24 +1,21 @@
 package org.jboss.da.reports.backend.impl;
 
-import java.lang.invoke.MethodHandles;
-import java.util.ArrayList;
-
+import org.jboss.da.communication.CommunicationException;
```
Comment (janinko) [inline reports-backend/src/main/java/org/jboss/da/reports/backend/impl/VersionFinderImpl.java]: Just changed order.

File: reports-backend/src/main/java/org/jboss/da/reports/backend/impl/VersionFinderImpl.java [hunk 6]
```diff
@@ -1,24 +1,21 @@
...
 import org.jboss.da.communication.aprox.api.AproxConnector;
-import org.jboss.da.communication.aprox.model.GA;
-import org.jboss.da.communication.aprox.model.GAV;
+import org.jboss.da.communication.model.GAV;
 import org.jboss.da.reports.backend.api.VersionFinder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (janinko) [inline reports-backend/src/main/java/org/jboss/da/reports/backend/impl/VersionFinderImpl.java]: Just changed order.

File: reports-backend/src/main/java/org/jboss/da/reports/backend/impl/VersionFinderImpl.java [hunk 7]
```diff
@@ -1,24 +1,21 @@
...
 
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
 
+import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
```
Comment (janinko) [inline reports-backend/src/main/java/org/jboss/da/reports/backend/impl/VersionFinderImpl.java]: Just changed order.

File: reports-backend/src/test/java/org/jboss/da/reports/backend/impl/VersionFinderTest.java [hunk 8]
```diff
@@ -1,11 +1,12 @@
 package org.jboss.da.reports.backend.impl;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
 import static org.mockito.Mockito.when;
 
+import org.jboss.da.communication.CommunicationException;
```
Comment (janinko) [inline reports-backend/src/test/java/org/jboss/da/reports/backend/impl/VersionFinderTest.java]: Just changed order.

File: reports-rest/src/main/java/org/jboss/da/rest/reports/Reports.java [hunk 9]
```diff
@@ -19,6 +17,9 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+
```
Comment (janinko) [inline reports-rest/src/main/java/org/jboss/da/rest/reports/Reports.java]: Just changed order.

File: reports-rest/src/main/java/org/jboss/da/rest/reports/model/LookupReport.java [hunk 10]
```diff
@@ -1,9 +1,10 @@
 package org.jboss.da.rest.reports.model;
 
-import org.jboss.da.communication.aprox.model.GAV;
+import org.jboss.da.communication.model.GAV;
+
+import lombok.AllArgsConstructor;
```
Comment (janinko) [inline reports-rest/src/main/java/org/jboss/da/rest/reports/model/LookupReport.java]: Just changed order.

File: testsuite/src/test/java/org/jboss/da/test/communication/AproxTest.java [hunk 11]
```diff
@@ -2,21 +2,21 @@
...
 import org.jboss.da.test.AbstractArquillianTest;
 import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
+import javax.inject.Inject;
+
+import java.util.Arrays;
+import java.util.List;
+
```
Comment (janinko) [inline testsuite/src/test/java/org/jboss/da/test/communication/AproxTest.java]: Just changed order.

**Comment by jbartece** [comment]
We will use this order of imports:

org
javax
java
com

File: testsuite/src/test/java/org/jboss/da/test/communication/AproxTest.java (hunk 11 above)
Comment (jbartece) [inline testsuite/src/test/java/org/jboss/da/test/communication/AproxTest.java]: btw. it is not just changed order. There is changed import of GA class

**Pull request merged**